### ISSUE TITLE
DataFrames in AnnData.uns get stored as matrices

### ISSUE THREAD

**Issue by ndrubins** [theislab/anndata#403]
**EDIT**: real bug description starts at https://github.com/theislab/anndata/issues/403#issuecomment-664047572

Hi,

I'm using `scanpy` from `R` with `reticulate` and I'm creating several `R` `data.frame`s  (e.g. `my_df` below) that I would like to add to the `AnnData` object (e.g., under `obsm`) and save that modified `AnnData` as an `h5ad` file.

If for example, I read an existing `AnnData` h5ad file (`in.h5ad.fn` below) in `R` with `reticulate` using:
```
library(reticulate)
sc <- import("scanpy")
adata <- sc$read(in.h5ad.fn)

```

Then created:
`my_df <- data.frame(cell = rownames(adata$obs), value = rnorm(nrow(adata$obs))
`

I thought that:
`adata$obsm['my_df'] <- my_df
`

would work but I get this error:
```
Error in adata$obsm["my_df"] = my_df :
  object of type 'environment' is not subsettable
 
```

What's the proper way to add `my_df` to `adata`?
And what would be the command to save that modified `adata` object to an `h5ad` file?

**Comment by ndrubins** [comment]
The `anndata` and `scanpy` versions I was using are:
 
```
>>> import scanpy as sc
>>> print(sc.__version__)
1.4.4.post1
>>> import anndata as ad
>>> print(ad.__version__)
0.6.22.post1

```
So for `anndata` it is the latest [available on `anaconda`](https://anaconda.org/bioconda/anndata) but not so for [`scanpy` ](https://anaconda.org/bioconda/scanpy).

I did try to update my `scanpy` but now I'm getting an error when trying to import it which seems to relate to `matplotlb` -> `cairo` issues. (the error I'm getting for `import scanpy` traces back to: `AttributeError: module 'cairo' has no attribute 'version_info'`)

**Comment by ivirshup** [comment]
@ndrubins, [`anndata` is on `conda` through conda-forge](https://anaconda.org/conda-forge/anndata). It looks like you're linking to an outdated version from bioconda. You should be able to install a more recent version with `conda install -c conda-forge anndata`.

**Comment by flying-sheep** [comment]
Yeah, I should have asked this from the start!

**Comment by ndrubins** [comment]
Thank you very much for all the help and I apologize for not being aware of the most up to date `conda` distribution location for these packages.

**Comment by ivirshup** [comment]
That's alright. I'm a little surprised conda was able to install that old `bioconda` version, especially since `bioconda` depends on `conda-forge`. Do you happen to know what command resulted in that?

**Comment by ndrubins** [comment]
For installing `anndata` from `bioconda` I used:
`conda install -c bioconda anndata
`
(I hope I'm understanding your question correctly).

But I also installed `scanpy` and `scipy` from `bioconda` but didn't pay attention if that re-installing `anndata` and if so which version.

BTW, about the `cairo` error I ran into when importing `scanpy`, which I mentioned above, installing `pycairo` solved it, in case that comes up frequently as an issue.
